src/state.rs

```diff
@@ -51,6 +51,55 @@ impl State {
     Ok(())
   }
 
+  pub(crate) fn load(&mut self, path: PathBuf) -> Result<()> {
+    match path.extension() {
+      Some(ext) if ext == "txt" => {
+        let content = fs::read_to_string(&path)?;
+
+        let lines = content.lines();
+
+        let width = lines
+          .clone()
+          .collect::<Vec<&str>>()
+          .first()
+          .unwrap_or(&"")
+          .len();
+
+        let height = lines.clone().count();
+
+        self.matrix = DMatrix::from_iterator(
+          width,
+          height,
+          lines
+            .map(|line| {
+              line
+                .chars()
+                .map(|c| Vector3::from_element(c.to_digit(2).unwrap_or(0) as u8))
+            })
+            .flatten(),
+        );
+      }
+      _ => {
```

Comment:
I think this can be simpler:

- iterate over length-3 chunks of the image data
- .map to a `Vector3<u8>`
- .collect into a `Vec<Vector3<u8>>`
- Use `Matrix::from_vec(width, height, data)` to construct the matrix in the wrong orientation
- calle .transpose to put it into the right orientation

Reply:
Is this simpler? I think `Matrix::from_iterator(height, width, iterator)` is better, since it's less verbose and we don't need to transpose the matrix ourselves.

Reply:
`from_iterator` sounds great. I think we will still need to transpose it, since the `from_iterator` expects the data in column-major order.

Reply:
We should have a test that loads a picture, and then calls `print` to assert that the text output has the right shape and contents.